### PR TITLE
RX LUA: make (SBUS) Failsafe Mode line visible

### PR DESCRIPTION
**Pull request by mha1**
Problem: using the LUA script to change from non-SBUS type serial protocols like CRSF to SBUS type protocols (SBUS,  Inv. SBUS, DJI) only showed the (SBUS) failsafe mode settings line after power cycling the receiver.

Solution: instantly hide or make failsafe mode settings visible depending on newly selected serial protocol.

Additional change: fix typo

Edit after discussion:
- rx LUA will show Protocol2 unconditional if rx is on ESP32 platform
- Failsafe Mode renamed to SBUS failsafe to avoid confusion with PWM failsafe.
- SBUS failsafe setting will also be visible unconditionally
- changed Protocol2 setting NONE to Off

### CODE REVIEW

**Comment by pkendall64** [comment]
For bonus points we could hide the serial protocol entries if they are not configured too!

**Comment by mha1** [comment]
ok

File: src/lib/LUA/rx_devLUA.cpp [hunk 1]
```diff
@@ -474,6 +484,54 @@ static void luaparamSetPower(struct luaPropertiesCommon* item, uint8_t arg)
...
+#endif
+  
+  return (prot0 == PROTOCOL_SBUS || prot0 == PROTOCOL_INVERTED_SBUS || prot0 == PROTOCOL_DJI_RS_PRO
+#if defined(PLATFORM_ESP32)          
+          || prot1 == PROTOCOL_SERIAL1_SBUS || prot1 == PROTOCOL_SERIAL1_INVERTED_SBUS || prot1 == PROTOCOL_SERIAL1_DJI_RS_PRO
+#endif
+        );
+}
+
+static bool hasSerialdefined(int hardwarePin, eServoOutputMode mode)
```
Comment (pkendall64) [inline src/lib/LUA/rx_devLUA.cpp:501]: big D, i.e. `hasSerialDefined`

Comment (mha1) [reply src/lib/LUA/rx_devLUA.cpp]: done

**Review by pkendall64 — APPROVED** [review]

**Review by CapnBry — APPROVED** [review]
Looks great! Clever solution to the whole show/hide thing and nice to not see the `Protocol1` item on ESP any more which will avoid a lot of questions I am sure.

**Pull request merged**